### PR TITLE
Remove user write pools

### PR DESCRIPTION
**Pull request by Sriep**
### Changes
Changes to support single write pool. 
Some syntax fixes for currency changes (int64 -> uint64)

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- 0chain:https://github.com/0chain/0chain/pull/1465
- gosdk:https://github.com/0chain/gosdk/pull/490
- system_test: https://github.com/0chain/system_test/pull/315
- zboxcli:https://github.com/0chain/zboxcli/pull/258
- zwalletcli:
- Other: ...

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | remove_write_pools  |
| zboxcli  | remove_write_pools  |
| zwalletcli  | staging  |
| 0chain  | remove_write_pools  |
| blobber  | remove_write_pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | remove_write_pools  |
| zboxcli  | remove_write_pool  |
| zwalletcli  | staging  |
| 0chain  | remove_write_pools  |
| blobber  | remove_write_pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | remove/writepool  |
| gosdk  | remove_write_pools  |
| zboxcli  | remove_write_pool  |
| zwalletcli  | staging  |
| 0chain  | remove_write_pools  |
| blobber  | remove_write_pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | remove/writepool  |
| gosdk  | remove_write_pools  |
| zboxcli  | remove_write_pool  |
| zwalletcli  | staging  |
| 0chain  | write_pools  |
| blobber  | remove_write_pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | remove/writepool  |
| gosdk  | remove_write_pools  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | write_pools  |
| blobber  | remove_write_pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | remove/writepool  |
| gosdk  | remove_write_pools  |
| zboxcli  | remove_write_pool  |
| zwalletcli  | staging  |
| 0chain  | write_pools  |
| blobber  | remove_write_pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: code/go/0chain.net/core/transaction/entity.go [hunk 1]
```diff
@@ -106,6 +106,7 @@ type StorageAllocation struct {
 	CCT            time.Duration        `json:"challenge_completion_time"`
 	TimeUnit       time.Duration        `json:"time_unit"`
 	IsImmutable    bool                 `json:"is_immutable"`
+	WritePool      int64                `json:"write_pool"`
```
Comment (peterlimg) [inline code/go/0chain.net/core/transaction/entity.go]: Use uint64 for coin balance

File: code/go/0chain.net/blobbercore/allocation/protocol.go [hunk 2]
```diff
@@ -211,45 +205,31 @@ func RequestReadPoolStat(clientID string) (*ReadPool, error) {
...
+	resp, err = transaction.MakeSCRestAPICall(transaction.STORAGE_CONTRACT_ADDRESS, "/allocation", params, chain.GetServerChain())
 	if err != nil {
 		return nil, fmt.Errorf("requesting write pools stat: %v", err)
 	}
 
-	var pss []*PoolStat
-	if err = json.Unmarshal(resp, &pss); err != nil {
+	var allocation = struct {
+		ID        string `json:"id"`
+		WritePool int64  `json:"write_pool"`
```
Comment (peterlimg) [inline code/go/0chain.net/blobbercore/allocation/protocol.go]: Use uint64 for balance

**Review by peterlimg — CHANGES_REQUESTED** [review]
Please fix the lint errors, thanks.

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | remove/writepool  |
| gosdk  | remove_write_pools  |
| zboxcli  | remove_write_pool  |
| zwalletcli  | staging  |
| 0chain  | remove_write_pool  |
| blobber  | remove_write_pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | remove_write_pools  |
| zboxcli  | remove_write_pool  |
| zwalletcli  | staging  |
| 0chain  | write_pools  |
| blobber  | remove_write_pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by peterlimg — APPROVED** [review]
lgtm, please fix the conflicts and system tests

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | remove_write_pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | remove/writepool  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | remove_write_pool  |
| zwalletcli  | staging  |
| 0chain  | write_pools  |
| blobber  | remove_write_pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | remove/writepool  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | remove_write_pool  |
| zwalletcli  | staging  |
| 0chain  | write_pools  |
| blobber  | remove_write_pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | remove_write_pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**